docs/docs/unit-testing.md

```diff
@@ -0,0 +1,322 @@
+---
+title: Unit testing
+---
+
+Unit testing is a great way to protect against errors in your code before you
+deploy it. While Gatsby does not include support for unit testing out of the
+box, it only takes a few steps to get up and running. However there are a few
+features of the Gatsby build process that mean the standard Jest setup doesn't
+quite work. This guide shows you how to set it up.
+
+## Setting up your environment
+
+The most popular testing framework for React is [Jest](https://jestjs.io/),
+which was created by Facebook. While Jest is a general purpose JavaScript unit
+testing framework, it has lots of features that make it work particularly well
+with React.
+
+For this guide, you will be starting with `gatsby-starter-blog`, but the
+concepts should be the same or very similar for your site.
+
+First you need to install Jest and some more required packages. You need to
+install Babel 7 as it's required by Jest.
+
+```sh
+npm install --save-dev jest babel-jest react-test-renderer identity-obj-proxy 'babel-core@^7.0.0-0' @babel/core @babel/preset-env @babel/preset-react @babel/plugin-proposal-class-properties @babel/plugin-proposal-optional-chaining
+```
+
+Because Gatsby handles its own Babel configuration, you will need to manually
+tell Jest to use `babel-jest`. The easiest way to do this is to add a `"jest"`
+section in your `package.json`. You can set up some useful defaults at the same
+time:
+
+```json
+  "jest": {
+    "transform": {
+      "^.+\\.jsx?$": "<rootDir>/jest-preprocess.js"
+    },
+    "testRegex": "/.*(__tests__\\/.*)|(.*(test|spec))\\.jsx?$",
+    "moduleNameMapper": {
+      ".+\\.(css|styl|less|sass|scss)$": "identity-obj-proxy",
+      ".+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js"
+    },
+    "testPathIgnorePatterns": ["node_modules", ".cache"],
+    "transformIgnorePatterns": [
+      "node_modules/(?!(gatsby)/)"
+    ],
+    "globals": {
+      "__PATH_PREFIX__": ""
+    },
```

Comment:
I received the following error:
https://github.com/facebook/jest/issues/6766

`testURL: 'http://localhost'` solves this issue 👍

Reply:
Good spot. Added, along with explanation.